flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -816,6 +817,9 @@ public void attachJobGraph(List<JobVertex> topologiallySorted) throws JobExcepti
 
 		final ArrayList<ExecutionJobVertex> newExecJobVertices = new ArrayList<>(topologiallySorted.size());
 		final long createTimestamp = System.currentTimeMillis();
+		int maxPriorAttemptsHistoryLength = jobInformation.getJobConfiguration().getInteger(
```

Comment:
This is also the job's configuration and not the cluster configuration you are using here.

Reply:
@tillrohrmann, I not only use run junit tests to validate, but also run a single flink job to verify that it is correct today. May be wrong somewhere, I'll check it as your suggestion. thanks.

Reply:
hi @tillrohrmann, we need to discuss about the `maxPriorAttemptsHistoryLength` option where should we get it? 

**First,**, the option is used for `ExecutionJobVertex`, so it should be extracted from job's configuration. The cluster configuration is like `jobmanager.rpc.address`, `jobmanager.rpc.port` and so on.

**Second**, the key of the option is `jobmanager.execution.attempts-history-size`, I think it's better to be changed to `job.execution.attempts-history-size`. because the option is not used for `jobmanager`

Reply:
I think the maximum prior attempts history length is a cluster configuration option because it affects cluster operations. Therefore, it should come from the cluster's configuration which is accessible when you construct the `ExecutionGraph` in the `ExecutionGraphBuilder`.

Reply:
I would not change the name of the config option atm.

Reply:
@tillrohrmann, thanks for your detail explain. I'll think about it again.

Reply:
@tillrohrmann, hi, I have question that `ExecutionGraphBuilder.buildGraph` was called from `JobMaster`, but there is only one `jobMasterConfiguration` in JobMaster class. Can jobMasterConfiguration represent ClusterConfigration?

Reply:
yes it does

Reply:
> yes it does

done.